### PR TITLE
Reducing the time compexity

### PR DESCRIPTION
**Pull request by SalilVishnuKapur**
Here in I have added an elaborate comment describing what all is being done. Next I have reduced the number of if loops making it more concise and efficient.

### CODE REVIEW

File: sympy/sets/sets.py [hunk 1]
```diff
@@ -725,17 +725,12 @@ def __new__(cls, start, end, left_open=False, right_open=False):
         if not all(i.is_real is not False or i in inftys for i in (start, end)):
             raise ValueError("Non-real intervals are not supported")
 
-        # evaluate if possible
-        if (end < start) == True:
+        # evaluate if it is even possible to calculate the interval i.e start<less and if start==end then check the  
+        if ((end < start) == True) or ((end - start).is_negative) or (end == start and (left_open or right_open)):
```
Comment (moorepants) [inline sympy/sets/sets.py]: Please wrap lines to 79 characters.

**Comment by moorepants** [comment]
There are test failures that need to be addressed.

**Comment by gxyd** [comment]
I don't think this reduces time complexity. Actually we're not in a loop, but just making basic comparisons. I'm -1 to this PR.

**Comment by gxyd** [comment]
Closing this.

**Pull request closed without merging**